synapse/config/emailconfig.py

```diff
@@ -146,28 +143,45 @@ def read_config(self, config, **kwargs):
                     % (", ".join(missing),)
                 )
 
-            # Templates for password reset emails
+            # These email templates have placeholders in them, and thus must be
+            # parsed using a templating engine during a request
             self.email_password_reset_template_html = email_config.get(
                 "password_reset_template_html", "password_reset.html"
             )
             self.email_password_reset_template_text = email_config.get(
                 "password_reset_template_text", "password_reset.txt"
             )
+            self.email_registration_template_html = email_config.get(
+                "registration_template_html", "registration.html"
```

Comment:
IIRC we have to add new templates to UPGRADE.rst (and point the changelog to it), so that people who use a custom template_dir don't get messed up by it.

Reply:
Oh, hrm. Did not do that for password resets: https://github.com/matrix-org/synapse/blob/master/UPGRADE.rst#email

Should I add it retroactively in this PR as well as those for v1.4.0 templates?

Reply:
yeahh, I think that would be best.